core/citrus-base/src/main/java/org/citrusframework/actions/ReceiveMessageAction.java

```diff
@@ -217,18 +224,20 @@ protected void validateMessage(Message message, TestContext context) {
         }
 
         Message controlMessage = createControlMessage(context, messageType);
-        if (StringUtils.hasText(controlMessage.getName())) {
+        if (hasText(controlMessage.getName())) {
             context.getMessageStore().storeMessage(controlMessage.getName(), message);
         } else {
             context.getMessageStore().storeMessage(context.getMessageStore().constructMessageName(this, getOrCreateEndpoint(context)), message);
         }
 
-        if (validationProcessor != null) {
+        if (nonNull(validationProcessor)) {
             validationProcessor.validate(message, context);
         } else {
             logger.debug("Control message:\n{}", controlMessage.print(context));
 
-            if (!validators.isEmpty()) {
+            assumeMessageType(message);
```

Comment:
actually, I think it should be on `controlMessage` if it exists.

```suggestion
            if (nonNull(controlMessage)) {
              assumeMessageType(controlMessage);
            } else {
              assumeMessageType(message);
            }
```